nil/services/nilservice/service.go

```diff
@@ -369,6 +365,17 @@ func CreateNode(ctx context.Context, name string, cfg *Config, database db.DB, i
 			}
 		}
 
+		if cfg.ZeroState.MainKeys.MainPrivateKey != "" && cfg.ZeroState.MainKeys.MainPublicKey != "" {
+			execution.MainPrivateKey, err = crypto.HexToECDSA(cfg.ZeroState.MainKeys.MainPrivateKey)
```

Comment:
Hell, I think mutable global variable is the root of our problems.
Since there is no normal constant semantics in freaking Go (like in C++), these variables are basically `var`. But they were "logical" constants, initialized in `init()`. Now we start actually rewriting them in a completely different part of the program 🤯 Honestly, I think we should get rid of such constants (now it's not even a "singleton", it's even worse!) and fix the whole code to explicitly pass the corresponding values.

Reply:
I'm agree with that. Still this PR mostly about generating mainKeys inside nil gen config and we can't properly start archive nodes without that. So get rid of global variables take some time and I think it's better to make it in another PR.